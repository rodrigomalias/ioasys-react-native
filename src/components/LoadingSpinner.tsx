import React from "react"
import { ActivityIndicator, Dimensions, StyleSheet, View } from "react-native"
import { connect } from "react-redux"
import { setIsLoadingSpinner } from "../store/modules/loading/Action"
import { ILoadingReducer } from "../store/modules/loading/State"
import { appReducers } from "../store/Store"

export interface LoadingSpinnerProps {

}

function LoadingSpinner(props: LoadingSpinnerProps & ILoadingReducer) {
    const { isLoadingSpinner,  } = props
    return (
        isLoadingSpinner ? (
            <View style={styles.loadSpinner}>
                <ActivityIndicator size="large" color={'#01814e'} />
            </View>
        ) : (
            <></>
        )
    )
}

const isLoadingReducerData = ({ isLoadingReducer }: typeof appReducers) => {
    return {
        ...isLoadingReducer,
    }
}

export default connect(isLoadingReducerData, {
    setIsLoadingSpinner,
})(LoadingSpinner)

const styles = StyleSheet.create({
    loadSpinner: {
      height: Dimensions.get('window').height,
      justifyContent: 'center',
    },
    totalHeroesContainer: {
      flex: 1,
      padding: 20,
      alignItems: 'center',
    },
    totalHeroes: {
      fontWeight: 'bold',
      fontSize: 16,
    },
})