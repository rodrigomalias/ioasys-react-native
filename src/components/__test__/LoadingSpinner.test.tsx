import React from "react"
import renderer from "react-test-renderer"
import LoadingSpinner from "../LoadingSpinner"
import { render } from "@testing-library/react-native"

describe("<LoadingSpinner/>", () => {

    test("renders correctly", () => {
        const tree = renderer.create(<LoadingSpinner/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    test("should create an item", () => {
        const { getByTestId, } = render(<LoadingSpinner loading={true} />)

        const element = getByTestId ("ActivityIndicator")
        expect(element).toBeDefined()
    })
})