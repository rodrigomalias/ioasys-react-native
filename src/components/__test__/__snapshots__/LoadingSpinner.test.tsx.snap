// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner/> renders correctly 1`] = `null`;
